jetstream_pt/engine.py

```diff
@@ -447,6 +456,45 @@ def insert(
         slot,
     )
 
+  def precompute_ragged_block_indices(self, decode_state: DecodeState):
```

Comment:
The compute logic is complex, can you add high level function description to describe the logic?

Reply:
Good point! Pushed the commit with descriptions.